packages/types/custom/core.ts

```diff
@@ -665,6 +666,7 @@ export interface CoreGetters {
   hovered: Feature | null
   errors: PolarError[]
   map: Map
+  mapHasDimensions: boolean | null
```

Comment:
```suggestion
  mapHasDimensions: CoreState['mapHasDimensions']
```

Since `extends` doesn't work without a pretty long `Omit` here, I suggest we use this kind of reference to avoid duplications.

Reply:
This deviates from the current procedure, as other properties may also be the same as a State Parameter and as a Getter.

If you deem this necessary, we should go through all properties and add a reference to the state.

Reply:
I've checked how we do this otherwise, and in the draw plugin, we're actually using `extends` with `Omit` (and everywhere else, to avoid duplication, `extends`). I can't seem to write this as a regular suggestion as many of the affected lines have not been changed by this MR, but I suggest his:

```ts
export interface CoreGetters
  extends Omit<
    CoreState,
    | 'components'
    | 'center'
    | 'hovered'
    | 'map'
    | 'moveHandle'
    | 'moveHandleActionButton'
    | 'selected'
  > {
  // omitted from CoreState as actual getter type diverges
  components: PluginContainer[]
  hovered: Feature | null
  map: Map
  moveHandle: MoveHandleProperties
  moveHandleActionButton: MoveHandleActionButton
  selected: Feature | null

  // regular getters
  hasSmallHeight: boolean
  hasSmallWidth: boolean
  /** Whether the application currently has the same size as the visual viewport of the users browser */
  hasWindowSize: boolean
}
```

Deduplication and clearly marked type changes (i.e. vuex trickery). Do you like it?

Reply:
This seems to be a reasonable approach!

As this has nothing to do with the fix, do you agree that we should add this in another PR?